src/mlpack/core/metrics/non_maximal_supression_impl.hpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file nms_metric_impl.hpp
+ * @author Kartik Dutt
+ *
+ * Implementation of Non Maximal Supression metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_METRICS_NMS_IMPL_HPP
+#define MLPACK_CORE_METRICS_NMS_IMPL_HPP
+
+// In case it hasn't been included.
+#include "non_maximal_supression.hpp"
+
+namespace mlpack {
+namespace metric {
+
+template<bool UseCoordinates>
+template<
+    typename BoundingBoxesType,
+    typename ConfidenceScoreType,
+    typename OutputType
+>
+void NMS<UseCoordinates>::Evaluate(
+    const BoundingBoxesType& boundingBoxes,
+    const ConfidenceScoreType& confidenceScores,
+    OutputType& selectedBoundingBoxes,
+    const double threshold)
+{
+  Log::Assert(boundingBoxes.n_rows == 4, "Bounding boxes must be \
+      must contain only 4 rows determining coordinates of bounding \
+      box either in {x1, y1, x2, y2} or {x1, y1, h, w} format.\
```

Comment:
I think you intended to say `Bounding boxes must contain ...`

Reply:
Ahh, That's definitely a typo. Fixing it now. Thanks a lot.